### PR TITLE
Bootstrap Unit Testing

### PR DESCRIPTION
**Pull request by azaleacolburn**
### Description
This is a branch off of #1175 which I can actually edit to make fixes, since @ariesninjadev was working from a personal fork.

The tests are not my work, I have just fixed the formatting and typing issues.

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1942)

### CODE REVIEW

**Review by Dhruv-0-Arora — APPROVED** [review]

File: fission/src/test/bootstrap/SystemIntegration.test.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    //         pushState: vi.fn(),
+    //     }
+    //     Object.defineProperty(window, "history", {
+    //         value: mockHistory,
+    //         configurable: true,
+    //     })
+    //
+    //     expect(window.history.replaceState).toBeDefined()
+    //     expect(window.history.pushState).toBeDefined()
+    // })
```
Comment (Dhruv-0-Arora) [inline fission/src/test/bootstrap/SystemIntegration.test.tsx]: Note

**Comment by AlexD717** [comment]
It appears that you have multiple unit tests that test practically the same thing, particularly in main.test.tsx.
SameTest

File: fission/src/test/bootstrap/main.test.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+        expect(globalThis.gtag).toBeDefined()
+        expect(window.convertAuthToken).toBeDefined()
+    })
+
+    test("module is compatible with test environment", async () => {
+        await expect(import("@/main.tsx")).resolves.toBeDefined()
+
+        expect(window.convertAuthToken).toBeDefined()
+    })
```
Comment (BrandonPacewic) [inline fission/src/test/bootstrap/main.test.tsx]: Drawing attention to this so I know we made a decision before we merge.

What's the justification for having these tests that appear so similar?

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]

File: fission/src/test/bootstrap/main.test.tsx (hunk 2 above)
Comment (azaleacolburn) [reply fission/src/test/bootstrap/main.test.tsx]: I mean, he did use AI to write ~50% of the tests (according to him)

Comment (azaleacolburn) [reply fission/src/test/bootstrap/main.test.tsx]: In my opinion, we should remove the "handles DOM queries safely" test, because it definitely doesn't test DOM queries (although tbh, we should reevaluate the necessity of each of these tests).

Comment (Dhruv-0-Arora) [reply fission/src/test/bootstrap/main.test.tsx]: in my opinion we should just delete all of:

```ts
test("handles DOM queries safely", async () => {
        await expect(import("@/main.tsx")).resolves.toBeDefined()

        expect(globalThis.gtag).toBeDefined()
        expect(window.convertAuthToken).toBeDefined()
    })

    test("module is compatible with test environment", async () => {
        await expect(import("@/main.tsx")).resolves.toBeDefined()

        expect(window.convertAuthToken).toBeDefined()
    })
    ```

**Review by BrandonPacewic — COMMENTED** [review]
This might be the first case of AI "slop" within the Synthesis codebase, or at least in a PR. Going to combine a bunch of these tests into a single file as I'm seeing a lot of overlap here, should significantly cut down on the number of lines added by this PR.

File: fission/src/test/bootstrap/main.test.tsx (hunk 2 above)
Comment (BrandonPacewic) [reply fission/src/test/bootstrap/main.test.tsx]: Will clean this up.

Comment (itspvty) [reply fission/src/test/bootstrap/main.test.tsx]: Ok, I regret creating this PR before revising the code, especially with the use of AI.

**Review by AlexD717 — CHANGES_REQUESTED** [review]
I still feel that there are some tests in Synthesis.test.tsx that could be merged into one test
Screenshot 2025-07-07 081955
Screenshot 2025-07-07 081939

**Review by AlexD717 — CHANGES_REQUESTED** [review]
You are calling expect(screen.getByTestId("main-hud")).toBeDefined() in 5 tests. This is probably unnecessary and I would remove it from all tests but the first one. (A very similar thing is happening with screen.getByTestId("scene").

**Comment by itspvty** [comment]
> You are calling expect(screen.getByTestId("main-hud")).toBeDefined() in 5 tests. This is probably unnecessary and I would remove it from all tests but the first one. (A very similar thing is happening with screen.getByTestId("scene").

Seems like these calls are part of verifying the outcome of the previously executed steps in each test?

**Comment by AlexD717** [comment]
> > You are calling expect(screen.getByTestId("main-hud")).toBeDefined() in 5 tests. This is probably unnecessary and I would remove it from all tests but the first one. (A very similar thing is happening with screen.getByTestId("scene").
> 
> Seems like these calls are part of verifying the outcome of the previously executed steps in each test?

In some test it is verifying that the previous steps are correct. However, your third test renders the same exact thing as the first test, with no custom parameters or changes, and then verifies what was already verified in the first test. It verifies a new thing (that container.firstChild is defined) but it doesn't need to very that scene or main hud was defined.

File: fission/src/global.d.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import type { Mock } from "vitest"
+
+declare global {
+    interface Global {
+        gtag: Mock
+        URL: Mock & typeof URL
+        testProperty: string
```
Comment (rutmanz) [inline fission/src/global.d.ts:7]: Why is anything being globally typed as `Mock`? I think it's fine to type as `Function` but `vitest` stuff should not be put in places that make it relevant to the rest of the codebase

File: fission/src/test/bootstrap/AppStartupIntegration.test.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    })
+
+    test("bootstraps main.tsx without error", async () => {
+        await expect(import("@/main.tsx")).resolves.toBeDefined()
+    })
+
+    test("wires up analytics & auth callbacks", async () => {
+        await import("@/main.tsx")
+        expect(typeof window.convertAuthToken).toBe("function")
+        expect(typeof globalThis.gtag).toBe("function")
```
Comment (rutmanz) [inline fission/src/test/bootstrap/AppStartupIntegration.test.tsx:34]: Aren't you defining both of these as mocks and they will therefore always be defined as functions?

File: fission/src/test/bootstrap/ContextProviders.test.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+                                <TestComponent />
+                            </ToastProvider>
+                        </PanelControlProvider>
+                    </ModalControlProvider>
+                </TooltipControlProvider>
+            </ThemeProvider>
+        )
+
+        expect(screen.getByTestId("test-component")).toBeDefined()
+        expect(screen.getByTestId("context-test")).toBeDefined()
```
Comment (rutmanz) [inline fission/src/test/bootstrap/ContextProviders.test.tsx:196]: This tests that they don't error but doesn't test that they're actually providing any context/they're doing what they're expected to do, so I'm not entirely sure what this test is helpful for

File: fission/src/test/bootstrap/ContextProviders.test.tsx [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+    test("theme provider applies CSS variables correctly", () => {
+        render(
+            <ThemeProvider initialThemeName="Default" themes={mockThemes} defaultTheme={mockTheme}>
+                <TestComponent />
+            </ThemeProvider>
+        )
+
+        // The theme provider should set CSS custom properties on document root
+        // We can't easily test this in JSDOM, but we can verify the component renders
+        expect(screen.getByTestId("test-component")).toBeDefined()
```
Comment (rutmanz) [inline fission/src/test/bootstrap/ContextProviders.test.tsx:208]: ???

So why does this test exist?

File: fission/src/test/bootstrap/ContextProviders.test.tsx [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+        }).not.toThrow()
+
+        // Test providers with minimal props
+        expect(() => {
+            render(
+                <ThemeProvider initialThemeName="Default" themes={mockThemes} defaultTheme={mockTheme}>
+                    <TestComponent />
+                </ThemeProvider>
+            )
+        }).not.toThrow()
```
Comment (rutmanz) [inline fission/src/test/bootstrap/ContextProviders.test.tsx:228]: ???

File: fission/src/test/bootstrap/ContextProviders.test.tsx [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+
+        // Test with empty theme object
```
Comment (rutmanz) [inline fission/src/test/bootstrap/ContextProviders.test.tsx:230]: I don't think this is actually a useful test to have if this isn't a scenario that ever occurs. It is, in my opinion, not useful to test edge cases by just checking for an error if those edge cases won't come up, and you can check if they do by just checking if the main app renders

File: fission/src/test/bootstrap/Synthesis.test.tsx [hunk 9]
```diff
@@ -0,0 +1,358 @@
...
+            return defaults[key] ?? false
+        }),
+        setGlobalPreference: vi.fn(),
+        savePreferences: vi.fn(),
+        loadPreferences: vi.fn(),
+        addEventListener: vi.fn(),
+    },
+}))
+
+vi.mock("@/components/Scene.tsx", () => ({
```
Comment (rutmanz) [inline fission/src/test/bootstrap/Synthesis.test.tsx:50]: Why are all of the components and other relevant files being mocked? Shouldn't we be testing that they all load and render without error?

**Review by rutmanz — CHANGES_REQUESTED** [review]
> This might be the first case of AI "slop" within the Synthesis codebase, or at least in a PR. Going to combine a bunch of these tests into a single file as I'm seeing a lot of overlap here, should significantly cut down on the number of lines added by this PR.

Strongly agree with this, it looks like a lot of the tests were created without intention and don't actually test the important parts

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]
This branch needs to be manually updated as there were naming changes that are not reflected in git conflicts that are causing the tests to fail.

**Comment by azaleacolburn** [comment]
I'm going to take over and clean up this PR (@ariesninjadev learned his lesson apparently)

**Comment by azaleacolburn** [comment]
After further consideration, I don't actually think there's much of value in this PR at all. I'm going to close it and start fresh on  this ticket.
